service/src/common/network_subgraph.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use reqwest::{header, Client, Url};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::query_processor::{QueryError, UnattestedQueryResult};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Response<T> {
+    pub result: T,
+    pub status: i64,
+}
+
+/// Network subgraph query wrapper
+///
+/// This is Arc internally, so it can be cloned and shared between threads.
+#[derive(Debug, Clone)]
+pub struct NetworkSubgraph {
+    client: Client, // it is Arc
+    network_subgraph_url: Arc<Url>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct GraphQLQuery {
+    pub query: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variables: Option<Value>,
+}
+
+impl NetworkSubgraph {
+    pub fn new(
+        graph_node_query_endpoint: Option<&str>,
+        deployment: Option<&str>,
+        network_subgraph_url: &str,
+    ) -> NetworkSubgraph {
+        //TODO: Check indexing status of the local network subgraph deployment
+        //if the deployment is healthy and synced, use local_network_subgraph_endpoint
+        let _local_network_subgraph_endpoint = match (graph_node_query_endpoint, deployment)
+        {
+            (Some(_endpoint), Some(id)) =>
+            Some(Url::parse(graph_node_query_endpoint.unwrap())
```

Comment:
I think you meant it to be like this:
```suggestion
            Some(Url::parse(_endpoint)
```

Reply:
yikes! thank u